numpy-stubs/__init__.pyi

```diff
@@ -179,16 +181,16 @@ class _flagsobj:
     def __getitem__(self, key: str) -> bool: ...
     def __setitem__(self, key: str, value: bool) -> None: ...
 
-class flatiter:
+class flatiter(Generic[_ArraySelf], Iterator[generic]):
```

Comment:
Once ``ndarray`` is a proper Generic we can get more specific about the iterators elements; for now it's set to ``np.generic``.